broadlink/__init__.py

```diff
@@ -1063,6 +1063,18 @@ def set_state(self, state):
         cmd = '{"pwr":%d}' % (1 if state == "ON" or state == 1 else 0)
         self.send_command(cmd)
 
+    def set_brightness(self, state):
+        cmd = '{"brightness":%d}' % state
+        self.send_command(cmd)
+
+    def set_power(self, state):
+        self.set_brightness(100)
+        self.set_state(state)
```

Comment:
I don't think this method is necessary.

Reply:
I wrote this as a hook to broadlink_cli . Am I wrong?

Reply:
I think you could change the name of the method instead. This class is new, I don't think there are many integrations depending on it. If something breaks, I'll help fix it.

Reply:
This hook is an anti-pattern. Interfaces should be unique. I think the best option is to change the name of the original method to make this class compatible with the others.

Reply:
This class needs a rework. set_state() should be renamed to set_power(). And the new set_state() should get a dictionary as input. I intend to rework this class when I find some time, but I am working on a lot of things, so feel free to address this issue if you want to (you will need to rebase or start over). I think we really need a better interface for this class.

Ah, we don't want to send two packets with the same function. Our communication must be atomic, so client applications can do their own way and they can handle exceptions better. set_power() shouldn't set_brightness(100).